### PR TITLE
Fixing mask_embed for latest changes?

### PR DESCRIPTION
**Pull request by ctr26**
@afoix 

- dropping batch_idx
- added shape tests for sanity checking
- class MaskEmbedSupervised(AutoEncoderSupervised, MaskEmbed):
     pass
- Adding prediction tests 
This is not implemented nor present in the new test

### CODE REVIEW

**Comment by ctr26** [comment]
@afoix I added a MaskEmbed mixin class, this is used to overwrite the two AutoEncoder functions (batch_to_tensor and loss_function) that are needed

File: bioimage_embed/shapes/lightning.py [hunk 1]
```diff
@@ -53,15 +54,24 @@ def loss_function(self, model_output, *args, **kwargs):
...
-        return loss
+        return {
+            "loss": loss,
+            "shape_loss": shape_loss,
+            "reconstruction_loss": model_output.recon_x,
+            "variational_loss": variational_loss,
+        }
+
+
+class MaskEmbed(AutoEncoderUnsupervised, MaskEmbedMixin):
```
Comment (afoix) [inline bioimage_embed/shapes/lightning.py:67]: Just checking my understanding of python's multiple inheritance here... browsing around (https://www.geeksforgeeks.org/multiple-inheritance-in-python/), it seems that the first (leftmost) parent class' method implementations are inherited when several parents share a method. I thought the point of `MaskEmbedMixin` was precisely to overwrite `batch_to_tensor` and `loss_function` from `AutoEncoder`. Should that line instead be:
`class MaskEmbed(MaskEmbedMixin, AutoEncoderUnsupervised):` ?

File: bioimage_embed/shapes/lightning.py [hunk 2]
```diff
@@ -53,15 +54,24 @@ def loss_function(self, model_output, *args, **kwargs):
...
+    def __init__(self, model, args=SimpleNamespace()):
+        super().__init__(model, args)
+
+
+class MaskEmbedSupervised(AutoEncoderSupervised, MaskEmbedMixin):
```
Comment (afoix) [inline bioimage_embed/shapes/lightning.py:72]: Same as above.

**Comment by afoix** [comment]
It's also probably worth exporting the new `MaskEmbedSupervised` class in `bioimage_embed/shapes/__init__.py`.

File: bioimage_embed/shapes/lightning.py (hunk 1 above)
Comment (ctr26) [reply bioimage_embed/shapes/lightning.py:67]: Yep, wrong way round will fix

**Comment by ctr26** [comment]
- Fixed the mixin
- Testing for 1 - many classes
- Reorganised frobenius_norm logic

**Pull request merged**